### PR TITLE
Catch unused API compatibility fixtures

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

* Adds HPA v2 to API round trip tests (was missed in 1.23), updates data to match what was actually shipped in 1.23
* Adds check for unused API compatibility fixtures
* Removes unused API compatibility fixtures

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/cc @thockin

### CODE REVIEW

File: staging/src/k8s.io/api/testdata/HEAD/autoscaling.v2.HorizontalPodAutoscaler.json [hunk 1]
```diff
@@ -279,4 +279,4 @@
       }
     ]
   }
-}
+}
```
Comment (thockin) [inline staging/src/k8s.io/api/testdata/HEAD/autoscaling.v2.HorizontalPodAutoscaler.json:282]: Is this something we care about?

I ran yamlfmt on these, for kicks, and it complained.

**Review by thockin — COMMENTED** [review]
/lgtm
/approve

/hold

in case you care about final newlines

File: staging/src/k8s.io/api/testdata/HEAD/autoscaling.v2.HorizontalPodAutoscaler.json (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/api/testdata/HEAD/autoscaling.v2.HorizontalPodAutoscaler.json:282]: this is the output of our encoder, so I'd rather use it verbatim

**Comment by thockin** [comment]
ACK

**Pull request merged**